context_lab/src/components/Clock/Clock.js

```diff
@@ -0,0 +1,25 @@
+import React, { useState, useEffect, useContext } from "react"
+import { ThemeContext } from "../../providers/ThemeProvider"
+
+const Clock = () => {
+    const [ clock , setClock ] = useState(0)
+    const { theme } = useContext(ThemeContext)
+    let time = new Date().toLocaleString() 
+
+    useEffect(() => {
+        let timer = setInterval(() => {
+            setClock(time)
+        }, 1000)
+
```

Comment:
remember to remove the event listener on the componentWillUnmount section of the useEffect, otherwise there will be a lot of setIntervals ex:
`return () => clearInterval(timer)`